plugins/ccgx/fu-ccgx-i2c.c

```diff
@@ -0,0 +1,774 @@
+	CY_I2C_GET_STATUS_CMD,		/* retreive the I2C bus status.
+					 * value = bit0 - 0: TX 1: RX, bit15 - scbIndex, length = 3,
+					 * data_in = byte0: bit0 - flag, bit1 -	bus_state, bit2 - SDA state,
+					 * bit3 - TX underflow, bit4 - arbitration error, bit5 - NAK
+					 * bit6 - bus error,
+					 * byte[2:1] Data count remaining */
+	CY_I2C_RESET_CMD,		/* the command cleans up the I2C state machine and frees the bus
+					 * value = bit0 - 0: TX path, 1: RX path; bit15 - scbIndex,
+					 * length = 0 */
+	CY_SPI_READ_WRITE_CMD =	0xCA,	/* the command starts a read / write operation at SPI
+					 * value = bit 0 - RX enable, bit 1 - TX enable, bit 15 - scbIndex;
+					 * index = length of transfer */
+	CY_SPI_RESET_CMD,		/* the command resets the SPI pipes and allows it to receive new
+					 * request
+					 * value = bit 15 - scbIndex */
+	CY_SPI_GET_STATUS_CMD,		/* the command returns the current transfer status
+					 * the count will match the TX pipe status at SPI end
+					 * for completion of read, read all data
+					 * at the USB end signifies the	end of transfer
+					 * value = bit 15 - scbIndex */
+	CY_JTAG_ENABLE_CMD = 0xD0,	/* enable JTAG module */
+	CY_JTAG_DISABLE_CMD,		/* disable JTAG module */
+	CY_JTAG_READ_CMD,		/* jtag read vendor command */
+	CY_JTAG_WRITE_CMD,		/* jtag write vendor command */
+	CY_GPIO_GET_CONFIG_CMD = 0xD8,	/* get the GPIO configuration */
+	CY_GPIO_SET_CONFIG_CMD,		/* set the GPIO configuration */
+	CY_GPIO_GET_VALUE_CMD,		/* get GPIO value */
+	CY_GPIO_SET_VALUE_CMD,		/* set the GPIO value */
+	CY_PROG_USER_FLASH_CMD = 0xE0,	/* program user flash area. The total space available is 512 bytes
+					 * this can be accessed by the user from USB. The flash	area
+					 * address offset is from 0x0000 to 0x00200 and can be written to
+					 * page wise (128 byte) */
+	CY_READ_USER_FLASH_CMD,		/* read user flash area. The total space available is 512 bytes
+					 * this	can be accessed by the user from USB. The flash	area
+					 * address offset is from 0x0000 to 0x00200 and can be written to
+					 * page wise (128 byte) */
+	CY_DEVICE_RESET_CMD = 0xE3,	/* performs a device reset from firmware */
+} CyVendorCommand;
+
+typedef struct __attribute__((packed)) {
+	guint32	 frequency;		/* frequency of operation. Only valid values are 100KHz and 400KHz */
+	guint8	 slave_address;		/* slave address to be used when in slave mode */
+	guint8	 is_msb_first;		/* whether to transmit most significant bit first */
+	guint8	 is_master;		/* whether to block is to be configured as a master*/
+	guint8	 s_ignore;		/* ignore general call in slave mode */
+	guint8	 is_clock_stretch;	/* wheteher to stretch clock in case of no FIFO	availability */
+	guint8	 is_loop_back;		/* whether to loop back	TX data to RX. Valid only for debug purposes */
+	guint8	reserved[6];		/* reserved for future use */
+} CyUsI2cConfig_t;
+
+static gboolean
+i2c_reset (FuDevice *device, CyI2CDeviceHandle *handle, guint8 mode, GError **error)
```

Comment:
this function name should have more to it.  fu_ccg_i2c_reset perhaps?

Reply:
static functions are used only in the file.
Do I need to add prefix (fu-ccgx-i2c) to static functions?
If yes, I will add prefix to static functions.

Reply:
I fixed it renaming i2c_reset to fu_ccgx_i2c_reset.